### PR TITLE
Allow configuring more URLs through CMake

### PR DESCRIPTION
**Pull request by Trial97**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -176,6 +176,8 @@ endif()
 set(Launcher_NEWS_RSS_URL "https://prismlauncher.org/feed/feed.xml" CACHE STRING "URL to fetch Prism Launcher's news RSS feed from.")
 set(Launcher_NEWS_OPEN_URL "https://prismlauncher.org/news" CACHE STRING "URL that gets opened when the user clicks 'More News'")
 set(Launcher_HELP_URL "https://prismlauncher.org/wiki/help-pages/%1" CACHE STRING "URL (with arg %1 to be substituted with page-id) that gets opened when the user requests help")
+set(Launcher_SUCCESSFULL_LOGIN_URL "https://prismlauncher.org/successful-login" CACHE STRING "URL that gets opened when the user succesfully logins.")
```
Comment (getchoo) [inline CMakeLists.txt]:
```suggestion
set(Launcher_LOGIN_CALLBACK_URL "https://prismlauncher.org/successful-login" CACHE STRING "URL that gets opened when the user successfully logins.")
```

File: buildconfig/BuildConfig.cpp.in [hunk 2]
```diff
@@ -116,16 +116,19 @@ Config::Config()
     NEWS_RSS_URL = "@Launcher_NEWS_RSS_URL@";
     NEWS_OPEN_URL = "@Launcher_NEWS_OPEN_URL@";
     HELP_URL = "@Launcher_HELP_URL@";
+    SUCCESSFULL_LOGIN_URL = "@Launcher_SUCCESSFULL_LOGIN_URL@";
```
Comment (getchoo) [inline buildconfig/BuildConfig.cpp.in]:
```suggestion
    LOGIN_CALLBACK_URL = "@Launcher_LOGIN_CALLBACK_URL@";
```

File: buildconfig/BuildConfig.h [hunk 3]
```diff
@@ -133,6 +133,11 @@ class Config {
      */
     QString HELP_URL;
 
+    /**
+     * URL that gets opened when the user succesfully logins.
+     */
+    QString SUCCESSFULL_LOGIN_URL;
```
Comment (getchoo) [inline buildconfig/BuildConfig.h]:
```suggestion
    QString LOGIN_CALLBACK_URL;
```

File: launcher/minecraft/auth/steps/MSAStep.cpp [hunk 4]
```diff
@@ -90,15 +90,16 @@ MSAStep::MSAStep(AccountData* data, bool silent) : AuthStep(data), m_silent(sile
...
+      <meta http-equiv="Refresh" content="0; URL=%1" />
     </noscript>
     Login Successful, redirecting...
     <script>
-      window.location.replace("https://prismlauncher.org/successful-login");
+      window.location.replace("%1");
     </script>
-    )XXX");
+    )XXX")
+                                          .arg(BuildConfig.SUCCESSFULL_LOGIN_URL));
```
Comment (getchoo) [inline launcher/minecraft/auth/steps/MSAStep.cpp]:
```suggestion
                                          .arg(BuildConfig.LOGIN_CALLBACK_URL));
```

**Review by getchoo — CHANGES_REQUESTED** [review]
I think this is a more proper oauth-y name

**Comment by getchoo** [comment]
This is also mainly an internal change, so I don't see much need for it in the changelog

File: CMakeLists.txt [hunk 5]
```diff
@@ -205,6 +207,7 @@ set(Launcher_BUG_TRACKER_URL "https://github.com/PrismLauncher/PrismLauncher/iss
 
 # Translations Platform URL
 set(Launcher_TRANSLATIONS_URL "https://hosted.weblate.org/projects/prismlauncher/launcher/" CACHE STRING "URL for the translations platform.")
+set(Launcher_TRANSLATIONS_BASE_URL "https://i18n.prismlauncher.org/" CACHE STRING "URL for the translations files.")
```
Comment (getchoo) [inline CMakeLists.txt]:
```suggestion
set(Launcher_TRANSLATION_FILES_URL "https://i18n.prismlauncher.org/" CACHE STRING "URL for the translations files.")
```
Using `BASE_URL` here makes me think it's related to the variable above, not what it's actually for

File: buildconfig/BuildConfig.cpp.in [hunk 6]
```diff
@@ -116,16 +116,19 @@ Config::Config()
...
     FLAME_API_KEY = "@Launcher_CURSEFORGE_API_KEY@";
     META_URL = "@Launcher_META_URL@";
+    FMLLIBS_BASE_URL = "@Launcher_FMLLIBS_BASE_URL@";
 
     GLFW_LIBRARY_NAME = "@Launcher_GLFW_LIBRARY_NAME@";
     OPENAL_LIBRARY_NAME = "@Launcher_OPENAL_LIBRARY_NAME@";
 
     BUG_TRACKER_URL = "@Launcher_BUG_TRACKER_URL@";
     TRANSLATIONS_URL = "@Launcher_TRANSLATIONS_URL@";
+    TRANSLATIONS_BASE_URL = "@Launcher_TRANSLATIONS_BASE_URL@";
```
Comment (getchoo) [inline buildconfig/BuildConfig.cpp.in]:
```suggestion
    TRANSLATION_FILES_URL = "@Launcher_TRANSLATION_FILES_URL@";
```
I'm fine with keeping this name internally though as it doesn't matter *too* much, but if you're up to `s/TRANSLATIONS_BASE_URL/TRANSLATION_FILES_URL/g` feel free

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]

**Review by Ryex — APPROVED** [review]

**Pull request merged**